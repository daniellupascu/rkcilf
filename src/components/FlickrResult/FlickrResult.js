import React from "react";
import PropTypes from "prop-types";

import Loader from "../Loader/Loader";
import { LoaderContainer } from "./FlickrResultStyled";
import FlickrGallery from "../FlickrGallery/FlickrGallery";

const FlickrResult = ({ isLoading, isNotFound, images }) => {
  return (
    <>
      {isLoading ? (
        <LoaderContainer>
          <Loader />
        </LoaderContainer>
      ) : isNotFound || images.length === 0 ? (
        <LoaderContainer>no images found</LoaderContainer>
      ) : images.length > 0 ? (
        <FlickrGallery images={images} />
      ) : null}
    </>
  );
};

FlickrResult.propTypes = {
  isLoading: PropTypes.bool,
  isNotFound: PropTypes.bool,
  images: PropTypes.arrayOf(PropTypes.object),
};

export default FlickrResult;
