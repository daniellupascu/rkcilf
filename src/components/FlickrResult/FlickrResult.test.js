import React from "react";
import renderer from "react-test-renderer";
import FlickrResult from "./FlickrResult";

describe("Test FlickrResult Component", () => {
  it("renders correctly", () => {
    const images = [
      {
        src: "url.com",
        title: { _text: "random Image" },
        author: { name: { _text: "Random User" } },
      },
      {
        src: "url.dk",
        title: { _text: "random Image" },
        author: { name: { _text: "Random User" } },
      },
    ];
    const tree = renderer
      .create(
        <FlickrResult images={images} isLoading={false} isNotFound={false} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly the loading state", () => {
    const tree = renderer
      .create(<FlickrResult images={[]} isLoading={true} isNotFound={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly the not found state", () => {
    const tree = renderer
      .create(<FlickrResult images={[]} isLoading={false} isNotFound={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
