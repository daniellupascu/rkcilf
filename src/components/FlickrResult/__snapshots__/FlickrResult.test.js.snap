// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FlickrResult Component renders correctly 1`] = `
<div
  className="FlickrGalleryStyled__FlickrGalleryContainer-sc-3z0gpy-0 kwOdIo"
>
  <figure
    className="FlickrGalleryStyled__FlickrImageContainer-sc-3z0gpy-1 kxaNwV"
  >
    <img
      className="FlickrGalleryStyled__FlickrImage-sc-3z0gpy-2 dxtjqM"
      src="url.com"
    />
    <div
      className="FlickrGalleryStyled__ImageInfo-sc-3z0gpy-3 bHTbjt"
    >
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        random Image
      </div>
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        credits: 
        Random User
      </div>
    </div>
  </figure>
  <figure
    className="FlickrGalleryStyled__FlickrImageContainer-sc-3z0gpy-1 kxaNwV"
  >
    <img
      className="FlickrGalleryStyled__FlickrImage-sc-3z0gpy-2 dxtjqM"
      src="url.dk"
    />
    <div
      className="FlickrGalleryStyled__ImageInfo-sc-3z0gpy-3 bHTbjt"
    >
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        random Image
      </div>
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        credits: 
        Random User
      </div>
    </div>
  </figure>
</div>
`;

exports[`Test FlickrResult Component renders correctly the loading state 1`] = `
<div
  className="sc-AxjAm khEyo"
>
  <svg
    height="44"
    stroke="#c179b9"
    viewBox="0 0 44 44"
    width="44"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
      strokeWidth="2"
    >
      <circle
        cx="22"
        cy="22"
        r="1"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="spline"
          dur="1.8s"
          keySplines="0.165, 0.84, 0.44, 1"
          keyTimes="0; 1"
          repeatCount="indefinite"
          values="1; 20"
        />
        <animate
          attributeName="stroke-opacity"
          begin="0s"
          calcMode="spline"
          dur="1.8s"
          keySplines="0.3, 0.61, 0.355, 1"
          keyTimes="0; 1"
          repeatCount="indefinite"
          values="1; 0"
        />
      </circle>
      <circle
        cx="22"
        cy="22"
        r="1"
      >
        <animate
          attributeName="r"
          begin="-0.9s"
          calcMode="spline"
          dur="1.8s"
          keySplines="0.165, 0.84, 0.44, 1"
          keyTimes="0; 1"
          repeatCount="indefinite"
          values="1; 20"
        />
        <animate
          attributeName="stroke-opacity"
          begin="-0.9s"
          calcMode="spline"
          dur="1.8s"
          keySplines="0.3, 0.61, 0.355, 1"
          keyTimes="0; 1"
          repeatCount="indefinite"
          values="1; 0"
        />
      </circle>
    </g>
  </svg>
</div>
`;

exports[`Test FlickrResult Component renders correctly the not found state 1`] = `
<div
  className="sc-AxjAm khEyo"
>
  no images found
</div>
`;
